### PR TITLE
Allow dynamic time based info to be included in file names

### PR DESCRIPTION
**Pull request by silverboots**
### Required for all PRs:

- [ ] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] README.md updated (if adding a new plugin)

So logs can be put to a dated file name / directory, support expression expansion in file name.  This allows files to be archived more easily and is a precursor to rolling logs based on day / size.

### CODE REVIEW

**Comment by silverboots** [comment]
Hi could I get some feedback n this pull request please.
Thanks

File: plugins/outputs/file/file.go [hunk 1]
```diff
@@ -21,8 +23,10 @@ type File struct {
 
 var sampleConfig = `
   ## Files to write to, "stdout" is a specially handled file.
+  ## Will also accept fromats such as /tmp/{020106}/metrics{020106.150406}.out
```
Comment (sparrc) [inline plugins/outputs/file/file.go]: spelling

Comment (sparrc) [inline plugins/outputs/file/file.go]: what do these numbers mean? how do users define custom formats? what characters are supported?

**Comment by silverboots** [comment]
fixed

**Comment by silverboots** [comment]
Do you have any preference for implementation of roll on time/ size? I was going to do a check on each write to determine if a roll is required.

File: plugins/outputs/file/file.go (hunk 1 above)
Comment (silverboots) [inline plugins/outputs/file/file.go]: They are the standard format options in go see https://golang.org/pkg/time/#Time.Format - so it supports anything go supports

Comment (sparrc) [inline plugins/outputs/file/file.go]: you will need to document that better and do a full writeup in the README with examples.

Comment (silverboots) [inline plugins/outputs/file/file.go]: Done

Comment (silverboots) [inline plugins/outputs/file/file.go]: Hi, are you waiting on me to do something on this ? - I believe I have covered everything requested, but if not, please let me know.

**Comment by silverboots** [comment]
Hi can you let me know if this is alright to accept now please.

File: plugins/outputs/file/file.go [hunk 2]
```diff
@@ -20,14 +22,21 @@ type File struct {
...
 var sampleConfig = `
-  ## Files to write to, "stdout" is a specially handled file.
-  files = ["stdout", "/tmp/metrics.out"]
-
-  ## Data format to output.
-  ## Each data format has it's own unique set of configuration options, read
-  ## more about them here:
-  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
-  data_format = "influx"
+	## Files to write to, "stdout" is a specially handled file.
```
Comment (sparrc) [inline plugins/outputs/file/file.go]: these should be indented by 2 spaces, not tabs

File: plugins/outputs/file/file.go [hunk 3]
```diff
@@ -20,14 +22,21 @@ type File struct {
...
+	## For files which contain curly bracket tokens, these tokens will be interpretted as a date/time format
```
Comment (sparrc) [inline plugins/outputs/file/file.go]: you need to respect the 80-char line length limit

Comment (sparrc) [inline plugins/outputs/file/file.go]: also please make this documentation a bit more brief, this is way too verbose.

**Comment by sparrc** [comment]
please be patient, I've been on vacation and then at gophercon

**Comment by silverboots** [comment]
all done, let me know if ok - hope you had a good holiday

**Comment by sparrc** [comment]
Sorry for the delay but I'm going to close this PR as a "won't merge", you will need to maintain a fork if you would like this functionality. The reason is that this change is highly specialized for your environment, and it adds a large amount of complication to the general user who doesn't care for this feature. What's more, this seems like something that an external service like `logrotate` should handle for you.

We will also eventually have support for dynamically reloading of configuration files, so another way you could implement this would be to just change the configuration file whenever you want a new file/directory for your metrics.

**Comment by silverboots** [comment]
With respect, I disagree with the close, but it is your decision. The change doesn't complicate usage as it will still work as before if required. Also, I don't think it's specialised to my usage, it's a generic requirement I have seen in many places. The suggestions you have provided appear to be a bit hacky in my opinion and I'm quite surprised by them. I appreciate file output is probably not a core scenario for you as there are far more sophisticated ones you have available, but it would have helped adoption for me and is currently very basic.

If you wish to discuss further let me know.

Thanks

On 19 Jul 2016, at 12:26, Cameron Sparr <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #1388<https://github.com/influxdata/telegraf/pull/1388>.

-

**Pull request closed without merging**